### PR TITLE
Adding discord Bot

### PR DESCRIPTION
**Pull request by yuvals2146**
discordBot is our way to crud the db for now 
it support the 4 commands 
1. `help`
2. `GetAllActivePositions`
3. `AddPosition`
4. `RemovePosition`

also added CI (sorry for all commits it was needed to test github flows) 
CI will run the tests for every push and will validate noting is broken.
in the future we'll create a docker image for every merge to main 

Bugs: it seems like Binance is blocked by us ISPs so a bypass was added to Binance test AND Binance calls to prevent  fails.

### CODE REVIEW

File: bin/test [hunk 1]
```diff
@@ -17,5 +17,5 @@ else
 # if SUITE_TO_RUN is not empty run a single test
     pp_info "running test: " "$SUITE_TO_RUN"
     pp_note "** if you wish to run all tests please remove SUITE_TO_RUN from .envrc.test **"
-    yarn jest --detectOpenHandles --no-cache --watchAll --testNamePattern=${SUITE_TO_RUN}
+    yarn jest --detectOpenHandles --no-cache --runInBand --watchAll --testNamePattern=${SUITE_TO_RUN}
```
Comment (azrield) [inline bin/test:20]: מה זה runInBand?

File: src/db/savePositionDataDB.js [hunk 2]
```diff
@@ -68,16 +68,35 @@ const userSaveNewPosition = async (position, txHash) => {
...
     const initData = await retriveInitalPositionData(position, txHash);
+    if (!initData) throw new Error("could not retrive initial data");
     saveInitialPositionInfo(position, initData);
   } catch (err) {
-    logger.error("error in userSaveNewPosition", err);
+    throw new Error(err.message);
   }
 };
+
+const deletePosition = async (position) => {
```
Comment (azrield) [inline src/db/savePositionDataDB.js:86]: Yeess

File: src/discordbot.js [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+    default:
+      msg.reply(
+        "I don't understand this command, try to use `help` to find posible commands"
+      );
+      break;
+  }
+});
+
+const getAllActivePositions = async (args) => {
```
Comment (azrield) [inline src/discordbot.js:70]: let's also accept ETH and ARB, and also case insensitive

File: src/discordbot.js [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+  return res;
+};
+
+const addPosition = async (args) => {
+  if (args.length < 2)
+    return "Need to supply at least position id and chain id";
+
+  const [positionChainName, positionId, txHash] = args;
+
+  if (positionChainName !== "ethereum" && positionChainName !== "arbitrum")
```
Comment (azrield) [inline src/discordbot.js:98]: same here

Comment (azrield) [reply src/discordbot.js:98]: very not important currently

File: src/test/discordBot.test.js [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+    test("should ask for help form bot", async () => {
+      await sendMsg(`<@${process.env.DISCORD_CLIENT_ID}> help`);
+      await new Promise((r) => setTimeout(r, sleep));
+      const response = await getRespose();
+      expect(response).toEqual(
+        "I can help you with the following commands: \n- `GetAllActivePositions` \n- `AddPosition` \n- `RemovePosition`"
+      );
+    });
+    test("should not recived help menu for diffrent keyword form bot", async () => {
+      await sendMsg(`<@${process.env.DISCORD_CLIENT_ID}> hell`);
```
Comment (azrield) [inline src/test/discordBot.test.js:44]: גדול

**Comment by azrield** [comment]
מעולה אתה! תודה :)

File: bin/test (hunk 1 above)
Comment (yuvals2146) [reply bin/test:20]: Make the test to *NOT* run in parallel

File: src/db/savePositionDataDB.js (hunk 2 above)
Comment (yuvals2146) [reply src/db/savePositionDataDB.js:86]: 🤠

File: src/blockchain/getPostionData.js [hunk 6]
```diff
@@ -569,33 +568,27 @@ const retriveInitalPositionData = async (position, txHash = null) => {
...
-        `No initial data found for position ${position.id} on chain ${
-          chains[position.chain].name
-        } please do it manually`
-      );
-      throw new Error(
-        `No inital data found for position ${position.id} on chain ${
-          chains[position.chain].name
-        }`
-      );
+      if (tx)
```
[the remark this answers is not in the document]
Comment (yuvals2146) [reply src/blockchain/getPostionData.js:571]: It is on creation, and the error will be catch send to discord as result

File: src/discordbot.js (hunk 3 above)
Comment (yuvals2146) [reply src/discordbot.js]: what do you mean?

Comment (yuvals2146) [reply src/discordbot.js]: Ha like 
 ethereum || ETH || eth ?

**Pull request merged**